tests/unit/test_clidriver.py

```diff
@@ -261,6 +261,15 @@ def test_error_logger(self):
         expected = {'log_level': logging.ERROR, 'logger_name': 'awscli'}
         self.assertEqual(driver.session.stream_logger_args[1], expected)
 
+    def test_ctrl_c_is_handled(self):
+        driver = CLIDriver(session=self.session)
+        fake_client = mock.Mock()
+        fake_client.list_objects.side_effect = KeyboardInterrupt
+        fake_client.can_paginate.return_value = False
+        driver.session.create_client = mock.Mock(return_value=fake_client)
+        rc = driver.main('s3 list-objects --bucket foo'.split())
+        self.assertEqual(rc, 130)
+
```

Comment:
Because this is a unittest, we'd never actually print the traceback to stderr.  The test runner would just catch the exception and log it in the error associated with the test result.

In order to test what you'd want I'd need to create an integration test, run an aws command, and send it a ctrl-c.

The only tricky part is that this is highly dependent on getting the timing right, we need to ctrl the process before it exists and after it's started and is making an API call.